### DISCUSSION TITLE
How use scipy's butter filter?

### DISCUSSION THREAD

**Discussion by ghost** [pydata/xarray#5749] [Q&A] [answered]
Hello friends,

I have SST data from 1870-2000 yearly. And I want to perform a low-pass filtering to this dataset.
Could anyone explain me that How should I use the "scipy.signal.butter" for low pass filtering my SST data for retaining the signal above 8 years? I searched for this:- scipy.signal.butter(N, Wn, btype='low', analog=False, output='ba', fs=None)
but what should I choose N, Wn particularly and other parameters for my case?

It would be helpful if someone could explain the procedure to do this. And if there is any other way then, please let me know.

Thank you.

**Comment by dcherian** [comment, marked as answer]
Hello,

Please use this forum for questions about xarray.

Maybe this scipy cookbook example helps: https://scipy-cookbook.readthedocs.io/items/ButterworthBandpass.html

or you can try this alpha-quality package: https://github.com/dcherian/xfilter

**Reply by ghost** [reply]
Thank you sir.
